feat: display uint props as known enum and bitfield types

flibdev opened the pull request:
# feat: Display `uint` props as known `enum` types

**Implemented:**
Some `uint` properties represent values of existing `enum` and `bitfield` types, but aren't stored as such. Rather than change the way they're stored or loaded, I added the `DisplayAsEnum` attribute and two new editor templates to render them as if they were those types.

Currently the following fields are affected:
- `CMaterialParameterInfo.Type`
- `MaterialTechnique.StreamsToBind`
- `MaterialTechnique.FeatureFlagsMask.Flags`
- `rendChunk.VertexFactory`

seberoth commented:
Just checked `WolvenKit.RED4` changes. Looks good to me. Can't test right now, though.

flibdev commented:
- Removed leftover restriction from `UIntAsEnumEditor` for values to be unsigned or non-zero (leftover from bitfield logic).
- Removed unused enum helper methods from DisplayAsEnumAttribute.
- Renamed editors to `DisplayAs___Editor`

manavortex commented on WolvenKit.RED4/Types/Classes/rendChunk.cs:
```diff
@@ -45,7 +45,8 @@ public CArray<CName> MaterialId
 		[Ordinal(5)] 
-		[RED("vertexFactory")] 
+		[RED("vertexFactory")]
+		[DisplayAsEnum<EMaterialVertexFactory>]
```
Should this maybe go into ClassesExt?

flibdev replied:
It could do, but I was worried about obfuscation and confusion if someone in the future wanted to know why the fields were rendered as they were. I just needed some way to pass info from the RedClasses down the pipeline to the renderer without hardcoding it, an attribute with a strongly-typed param seemed the cleanest way to do that

seberoth replied:
Its fine as is.
Doesn't make sense to put the attribute in ClassesExt

seberoth replied:
Only thing which is questionable is, if its really needed to do add a generic attribute.
